pkg/server/apiextensions.go

```diff
@@ -187,106 +253,119 @@ func (c *inheritanceCRDLister) GetWithContext(ctx context.Context, name string)
+
+	// TODO(ncdc): APIResourceSchema requires that its group be "" for the core group,
+	// See if we can unify things.
+	if group == "core" {
+		group = ""
 	}
 
-	// Check for API inheritance
-	targetWorkspaceKey := helper.WorkspaceKey(org, ws)
-	workspace, err := c.workspaceLister.Get(targetWorkspaceKey)
+	apiBindings, err := c.apiBindingLister.ListWithContext(ctx, labels.Everything())
 	if err != nil {
-		// If we're here it means ctx's logical cluster doesn't have the CRD and there isn't a
-		// ClusterWorkspace for the logical cluster. Just return not-found.
-		if apierrors.IsNotFound(err) {
-			return nil, apierrors.NewNotFound(apiextensionsv1.Resource("customresourcedefinitions"), name)
-		}
-
 		return nil, err
 	}
+	for _, apiBinding := range apiBindings {
+		if apiBinding.ClusterName != cluster.Name {
+			continue
+		}
+		if apiBinding.Status.Phase != apisv1alpha1.APIBindingPhaseBound {
+			// TODO(ncdc): what if it's in the middle of rebinding and it takes some time - we won't include any
+			// CRDs that were previously bound...
+			continue
+		}
 
-	if workspace.Spec.InheritFrom == "" {
-		// If we're here it means ctx's logical cluster doesn't have the CRD, the ClusterWorkspace exists,
-		// but it's not inheriting. Just return not-found.
-		return nil, apierrors.NewNotFound(apiextensionsv1.Resource("customresourcedefinitions"), name)
-	}
+		for _, boundResource := range apiBinding.Status.BoundResources {
+			if boundResource.Group == group && boundResource.Resource == resource {
+				crdKey := clusters.ToClusterAwareKey("system:bound-crds", boundResource.Schema.UID)
+				crd, err = c.crdLister.Get(crdKey)
+				if err != nil && !apierrors.IsNotFound(err) {
+					// something went wrong w/the lister - could only happen if meta.Accessor() fails on an item in the store.
+					return nil, err
+				}
 
-	var sourceWorkspaceCRDKey string
-	if workspace.Spec.InheritFrom == helper.RootCluster {
-		// HACK: allow inheriting from the OrganizationCluster logical cluster
-		sourceWorkspaceCRDKey = clusters.ToClusterAwareKey(helper.RootCluster, name)
-	} else {
-		// Make sure the source workspace exists
-		wsName := workspace.Spec.InheritFrom
-		orgClusterName := org
-		if strings.ContainsRune(workspace.Spec.InheritFrom, ':') {
-			orgClusterName, err = helper.ParentClusterName(workspace.Spec.InheritFrom)
-			if err != nil {
-				return nil, err
-			}
-			org, wsName, err = helper.ParseLogicalClusterName(workspace.Spec.InheritFrom)
-			if err != nil {
-				return nil, err
+				if crd != nil {
+					return crd, nil
+				}
+
+				// TODO(ncdc): if we got here, it means there is supposed to be a CRD coming from an APIBinding, but
+				// the CRD doesn't exist for some reason. Does it make sense to return a 404 here, or keep going and
+				// check other APIBindings and/or local CRDs?
+				return nil, apierrors.NewNotFound(schema.GroupResource{Group: apiextensionsv1.SchemeGroupVersion.Group, Resource: "customresourcedefinitions"}, name)
```

Comment:
404 is a definitive answer in kube, a consistent answer that this CRD does not exist. Wouldn't do that. This is merely some eventual-consistent state, i.e. return something that matches that, even a 50x would do. 404 is dangerous.